fix: panic in `regexp.MustCompile` when building wildcarddirectories with non ascii characters

camc314 opened the pull request:
Fixes #2236

jakebailey commented on internal/vfs/utilities.go:
```diff
@@ -81,11 +81,11 @@ func IsImplicitGlob(lastPathComponent string) bool {
-// It may be inefficient (we could just match (/[-[\]{}()*+?.,\\^$|#\s]/g), but this is future
-// proof.
+// Reserved characters - only escape actual regex metacharacters.
+// Go's regexp doesn't support \x escape sequences for arbitrary characters,
```
We shouldn't be using regexp at all; does switching to regexp2 fix this?

jakebailey replied:
It shouldn't be, given regexp2 has a mode that emulates ECMAScript regexes?

If we are going to use `regexp` and not `regexp2`, then we can probably just use `regexp.QuoteMeta`.

camc314 commented:
> The latest commit is not exactly what I mean; I wasn't trying to say to continue to change the escaping, but rather fix `getWildcardDirectories` to not use Go's regexp package, given the original code was more or less trying to emulate ECMAScript regexes which can be done with `regexp2`.
> 
> I feel somewhat less confident about hand escaping regexes like this unless we can be sure that those characters are definitely all that matter?
> 
> Otherwise, we'd just use https://pkg.go.dev/regexp#QuoteMeta, right?

Ah i see. i reverted that commit. The problem is that `GetRegularExpressionForWildcard` combines multiple strings, so it has to be escaped, hence we can't just omit the regex escaping change. Regardless, i changed `wildcarddirectories` to use `regexp2`.

Hopefully I understood that correctly 🙂

jakebailey commented:
I don't think the percent encoded thing is actually critical here, but you're definitely correct otherwise.
